### PR TITLE
Make plugin_data optional instead of repeated

### PR DESCRIPTION
**Pull request by chihuahua**
Every summary op writes data for a single plugin to process. Hence, each
SummaryMetadata proto should have a single PluginData optional field
(instead of a repeated one). This removes much complexity from
TensorBoard logic that loops over the plugin data. It also simplifies
the SQL schema - it can now enforce a one-to-one relationship between
summary op and plugin.

### CODE REVIEW

**Review by jart — APPROVED** [review]
@av8ramit Here is the 1.3 cherry-pick we discussed offline, which ensures a confusing `repeated` relationship in the `Summary` proto doesn't become a permanent part of TensorFlow's API.

**Comment by jart** [comment]
Mr. Jenkins: test this please.

**Comment by wchargin** [comment]
The included changes look good to me—are there really no changes to API goldens?

**Comment by chihuahua** [comment]
@wchargin, same question: `PASS: //tensorflow/tools/api/tests:api_compatibility_test`

**Comment by chihuahua** [comment]
Mr. Jenkins, test this please.

**Comment by chihuahua** [comment]
Mr. Jenkins: test this please.

**Comment by chihuahua** [comment]
Does anyone know why the tests don't run? Thank you!

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

File: tensorflow/python/summary/text_summary.py [hunk 1]
```diff
@@ -70,8 +70,8 @@ def text_summary(name, tensor, collections=None):
   summary_metadata = summary_pb2.SummaryMetadata()
   text_plugin_data = _TextPluginData()
   data_dict = text_plugin_data._asdict()  # pylint: disable=protected-access
-  summary_metadata.plugin_data.add(
-      plugin_name=PLUGIN_NAME, content=json.dumps(data_dict))
+  summary_metadata.plugin_data.plugin_name=PLUGIN_NAME
+  summary_metadata.plugin_data.content=json.dumps(data_dict))
```
Comment (av8ramit) [inline tensorflow/python/summary/text_summary.py]: Syntax error.

Comment (wchargin) [inline tensorflow/python/summary/text_summary.py]: Also lint errors (`s/=/ = /`).

Comment (chihuahua) [inline tensorflow/python/summary/text_summary.py]: Thank you!

Comment (chihuahua) [inline tensorflow/python/summary/text_summary.py]: Thank you!

**Comment by av8ramit** [comment]
Jenkins, test this please.

**Comment by av8ramit** [comment]
Jenkins, test this please.

**Comment by av8ramit** [comment]
Jenkins, test this please. Cmake build hung.

**Pull request merged**